src/EditorFeatures/Core/InheritanceMargin/InheritanceMarginServiceHelpers.cs

```diff
@@ -248,6 +272,36 @@ internal static class InheritanceMarginServiceHelper
                     .Concat(overridingMemberItems));
         }
 
+        private static ImmutableArray<ISymbol> GetImplementingSymbolsForTypeMember(
+            ISymbol memberSymbol,
+            ImmutableArray<ISymbol> overridingSymbols)
+        {
+            if (memberSymbol is IMethodSymbol or IEventSymbol or IPropertySymbol)
+            {
+                using var _ = ArrayBuilder<ISymbol>.GetInstance(out var builder);
+
+                // 1. Get the direct implementing symbols in interfaces.
+                var directImplementingSymbols = memberSymbol.ExplicitOrImplicitInterfaceImplementations();
+
+                // 2. Also add the direct implementing symbols for the overriding symbols.
+                // For example:
+                // interface IBar { void Foo(); }
+                // class Bar : IBar { public override void Foo() { } }
+                // class Bar2 : Bar { public override void Foo() { } }
+                // For 'Bar2.Foo()',  we need to find 'IBar.Foo()'
+                foreach (var symbol in overridingSymbols)
+                {
+                    builder.AddRange(symbol.ExplicitOrImplicitInterfaceImplementations()
+                        .Select(symbol => symbol.OriginalDefinition));
+                }
+
+                builder.AddRange(directImplementingSymbols.Select(symbol => symbol.OriginalDefinition));
```

Comment:
i guess my point is: have the helper method just return the symbols.  have the AddInheritanceMemberItemsForNamedTypeAsync be the one that calls each and *always* gets the original defs and ensure we only hae the distinct ones.

Reply:
: ) Got you this time